templates/master/00-master/_base/files/usr-local-bin-etcd-snapshot-backup-sh.yaml

```diff
@@ -17,18 +18,27 @@ contents:
     fi
 
     usage () {
-        echo 'Path to backup file required: ./etcd-snapshot-backup.sh ./backup.db'
+        echo 'Path to backup dir required: ./etcd-snapshot-backup.sh <path-to-backup-dir>'
         exit 1
     }
 
     ASSET_DIR=./assets
 
-    if [ "$1" != "" ]; then
-      SNAPSHOT_FILE="$1"
-    else
+    if [ -z "$1" ] || [ -f "$1" ]; then
       usage
     fi
 
+    if [ ! -d "$1" ]; then
+      mkdir -p $1
+    fi
+
+    BACKUP_DIR="$1"
+    DATESTRING=$(date "+%F_%H%M%S")
+    BACKUP_TAR_FILE=${BACKUP_DIR}/snapshot_db_kuberesources_$DATESTRING.tar
+    SNAPSHOT_FILE="${BACKUP_DIR}/snapshot.db"
```

Comment:
should both files have timestamp if we are going this route?

Reply:
Only the combined tar file bundle will have the timestamp. The snapshot.db needs to have this exact name so that we can extract it separately when restoring.

Reply:
how about creating a time-stamped directory inside the backup directory and putting the files in there?

something equivalent to
```
DATESTRING=$(date "+%F_%H%M%S")
BACKUP_DIR="$1/$DATESTRING"
mkdir -p BACKUP_DIR
```
This will make sure we don't overwrite the backups every time we run the script

Reply:
If the user wants he can give a backup directory that is timestamped. Since there is going to be a combined tar file, there is no question of overwriting the file. Let me modify to create the snapshot file in a temp directory and then append to the tar bundle.

Reply:
We ask the users to copy the snapshot file to other hosts. If we tar the snapshot along with the KAS assets are we going to ask users to copy KAS assets as well? I ask because I am not sure if KAS assets are node dependent or not, @retroflexer can you please confirm that?
 
IIRC the static pod yaml is in the assets directory which might differ from host to host in which case untaring the snapshot would just be an extra step a user has to do before scping for no added advantage.

Reply:
Yeah. I am not sure either.

Reply:
Checking with sttts.

Reply:
On my current installation, they appear to be identical.